### PR TITLE
Autocommit and push pre-commit changes in GHA

### PR DESCRIPTION
**Pull request by ralren**
## [Add the ability to commit and push changes made by terraform-fmt in the validate-tf GHA workflow](https://trello.com/c/s1jfNe0K)

Changes proposed in this pull request:

- See https://github.com/trussworks/terraform-aws-bootstrap/pull/109
- Add steps to pull PAT from 1Password and autocommit if precommit fails

### CODE REVIEW

File: .github/workflows/validate-tf.yml [hunk 1]
```diff
@@ -2,23 +2,47 @@ name: Validate - Terraform
...
           ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ env.PAT_SECRET }}
+
       - name: Render terraform docs and push changes back to PR
         uses: terraform-docs/gh-actions@main
         with:
           working-dir: .
           output-file: README.md
           output-method: inject
           git-push: "true"
```
[the remark this answers is not in the document]
Comment (ralren) [reply .github/workflows/validate-tf.yml:41]: which action are you referring to?

**Comment by rpdelaney** [comment]
We should test this on renovate PRs if at all possible. The terraform-docs action we have set up somewhere causes renovate to stop auto merging, because it does that any time anyone else pushed a commit to a PR it created

**Comment by ralren** [comment]
@rpdelaney you got a recommended repo that has renovatebot being very active?

**Comment by rpdelaney** [comment]
There's a pretty monstrous renovate PR in legendary-waddle right now that I've been dreading taking it on. If we merge this to `main` over there and then tell renovatebot to rebase, that should do the trick I think.

**Comment by ralren** [comment]
@rpdelaney worth a try!

**Pull request closed without merging**